### ISSUE TITLE
Performance speed issues?  (just a question)

### ISSUE THREAD

**Issue by ghost** [shmargum/vim-sass-colors#12]
In my case, this plugin obviously slows down the performance speed of vim. I don't know Vim well, but it seems to slow it down when it goes into buffer. It delays by about a second, when I move the window in `<C-w>hjkl` or when I use `:b` to move the When switching buffers. A similar plugin, colorizer, doesn't have this kind of delay. Is it just me? (I'm new to Vim so I don't know how to research it in detail. )

- macos 10.15.5
- vim 8.2.1100 using homebrew
- alacritty 0.4.3

**Comment by shmargum** [comment]
can you provide some more info or a more specific example here?  the plugin might be re-executing when context switching which I have at least partially addressed in the past

my vim workflow is usually pretty simple and i usually only have 1 buffer per window, but I did some fixes for vim split which has multiple buffers.. your workflow may be outside of my initial assumptions

**Comment by dadchyyy** [comment]
[Here's](https://github.com/dadchyyy/vim-sass-colors-sample) my configuration files and Sass files for reproducing the problem.

I use:
-MacBookAir2018
-macOSX 10.15.5
-alacritty 0.4.3
-tmux 3.1b ($TERM = screen-256color)
-vim 8.2.1100

[![Image from Gyazo](https://i.gyazo.com/d33d6bb82009c4b8a957315a1ea2c889.gif)](https://gyazo.com/d33d6bb82009c4b8a957315a1ea2c889)

Look at the GIF. There is a delay when opening the file, switching buffers, moving windows, and saving. If I don't use vim-sass-colors, there is no delay (It may be hard to see in the gif.).

**Comment by Scribler** [comment]
I'm also having this issue.  

I usually have 4 or more splits/buffers open at a time, and switching between them is slow, (slows more with additional buffers)
when I uninstall it goes back to normal.  I'm also pretty new to vim, so possibly there's something in my .vimrc that's messing it up.
-using vim plug for installation


[vimrc2.pdf](https://github.com/shmargum/vim-sass-colors/files/5543128/vimrc2.pdf)

**Comment by Scribler** [comment]
I added, "autocmd BufReadPre,BufNewFile * let b:did_vim_sass_colors = 1", and set the '1' to '0' The seems to have helped quite a bit.  There is still a little lag when switching between splits, but much better now.

**Comment by mrjones2014** [comment]
I also experience crippling performance issues with this plugin, even when just scrolling within the same file. I had to uninstall it because it was making nvim too slow.

**Comment by emmanuelclement** [comment]
Very interesting extension but unfortunately I had to uninstall it because of its very very slow speed.

It is indeed even slower when using several panels / split screen.

My projects use a lot of colour variables which are structured as follows:

A file of initial colour variables > A file of references to these variables > Final files using these references.

I would be happy to use this extension again if it improves its performance.

**Comment by mrjones2014** [comment]
@Scribler @emmanuelclement @shmargum for those wanting an alternative that works, there's https://github.com/norcalli/nvim-colorizer.lua. It is very fast and "just works"

**Comment by emmanuelclement** [comment]
@mrjones2014 : Yes, but the point of vim-sass-color is to tint the `$variables` themselves. As far as I know it is the only extension able to do this and it is its strength. (In the meantime I use vim-css-color.)
